base/applications/osk/main.c

```diff
@@ -359,6 +389,7 @@ INT_PTR APIENTRY OSK_DlgProc(HWND hDlg, UINT msg, WPARAM wParam, LPARAM lParam)
     {
         case WM_INITDIALOG:
             OSK_DlgInitDialog(hDlg);
+            OSK_SetAppIcon();
```

Comment:
You can directly call the load-icon + set-icon stuff within OSK_DlgInitDialog().